Add `Snapp Pay` driver

alissn opened the pull request:
Hi,

This pull request adds the `Snapp Pay` gateway, which facilitates installment payments. When creating a transaction, you need to pass the basket details to the gateway. The image below illustrates a simple basket with the required data.

![Basket Data](https://github.com/shetabit/multipay/assets/26966142/a69bd432-b932-4c01-aca8-7ba2c5f62fb9)

<div dir="rtl">

```
🔹گایدلاین ارسال دیتا در سرویس پیمنت توکن 🔹

➖مقادیر شامل در discountAmount :
کارت هدیه + کد تخفیف + امتیاز کاربر + کوپن

➖مقادیر شامل در externalSourceAmount :
کیف پول کاربر

➖ مقادیر شامل در shippingAmount :
بسته بندی + کادو پیچ + هزینه ارسال + هزینه بسته بندی محرمانه

➖ مقادیر شامل در totalAmount :
حاصل جمع مبالغ آیتم ها با در نظر گرفتن تعداد و هزینه tax و shipping
```
</div>

This driver includes the following methods:

- **eligible**: Ensures that you can offer the credit option to users.
- **settle**: Finalizes the payment after the `verify` method is called.
- **revert**: Refunds the payment to the user if the order cannot be completed (after verify).
- **status**: Retrieves the status of the payment.
- **cancel**: Cancels the payment (after settle).
- **update**: Changes the items in the basket, ensuring the final amount is less than the original amount.

Below is an image showing the process flow:

![Process Flow](https://github.com/shetabit/multipay/assets/26966142/0d9085d4-f839-4b7a-b60b-7a7d0e643ced)

### How has this been tested?
An account was set up in the staging environment of this gateway, and the following methods were tested:

- [x] `oauth/token`
- [x] `eligible`
- [x] `token`
- [x] `verify`
- [ ] `settle`
- [ ] `revert`
- [ ] `status`
- [ ] `cancel`
- [ ] `update`

Thanks.

khanzadimahdi commented on src/Drivers/SnappPay/SnappPay.php:
```diff
@@ -0,0 +1,454 @@
+        return $body['response'];
+    }
+
+    public function getPaymentUrl(): string
```
why this method is public? shouldn't it be private?

khanzadimahdi commented on src/Drivers/SnappPay/SnappPay.php:
```diff
@@ -0,0 +1,454 @@
+        return $this->paymentUrl;
+    }
+
+    public function setPaymentUrl(string $paymentUrl): void
```
also this one, why it is public?

alissn commented:
@khanzadimahdi

Thanks for the review. I have fixed the issues. Honestly, I copied the implementation from the `Digipay` gateway and didn't check the private/public status of those methods. :)

In #231, I also fixed the digipay driver.
